test/with_dummyserver/test_chunked_transfer.py

```diff
@@ -128,6 +147,34 @@ def test_remove_user_agent_header(self):
             ua_headers = self._get_header_lines(b"user-agent")
             assert len(ua_headers) == 0
 
+    def test_provides_default_transfer_encoding_header(self):
+        self.start_chunked_handler()
+        chunks = ["foo", "bar", "", "bazzzzzzzzzzzzzzzzzzzzzz"]
+        with HTTPConnectionPool(self.host, self.port, retries=False) as pool:
+            pool.urlopen("GET", "/", chunks, chunked=True)
+
+            te_headers = self._get_header_lines(b"transfer-encoding")
+            assert len(te_headers) == 1
+
+    def test_preserve_transfer_encoding_header(self):
+        self.start_chunked_handler()
+        chunks = ["foo", "bar", "", "bazzzzzzzzzzzzzzzzzzzzzz"]
+        with HTTPConnectionPool(self.host, self.port, retries=False) as pool:
+            pool.urlopen(
+                "GET",
+                "/",
+                chunks,
+                headers={"Transfer-Encoding": "test-transfer-encoding"},
```

Comment:
Can we use a non-standard casing here like `transfer-Encoding`?

Reply:
Done.